lib/releaselib.sh

```diff
@@ -50,12 +51,10 @@ release::update_job_cache () {
     logecho -r " $job cache..."
   fi
 
-  run=$($GSUTIL cat $logroot/$job/latest-build.txt 2>/dev/null) || return
-
   for ((cnt=0;cnt<=$cache_limit;cnt++)); do
-    # Once we hit one that is set, break out
+    # Once we hit a JOB[$run] that is set (or run gets to 0), break out
     # the cache should be complete after that
-    [[ -n ${JOB[$run]} ]] && break
+    ((run==0)) || [[ -n ${JOB[$run]} ]] && break
```

Comment:
On L47, we prefill the JOB dict with the existing cache.  Say the cache (dict) looks like this:
JOB[98]=somevalue
JOB[97]=somevalue

$run gets seeded with 100 from the latest-build.txt

In the loop we check for 2 conditions.  Has $run hit 0 or did we hit the first set item in the cache.
If those conditions aren't met, we set JOB[$run].
Then we decrement $run to 99.
Repeat
When JOB[98] is hit and prefilled ([[ -n ${JOB[$run]} ]]) we break.